include/rdma/fi_domain.h

```diff
@@ -185,6 +185,13 @@ enum fi_datatype {
 	FI_LONG_DOUBLE_COMPLEX,
 	/* End of point to point atomic datatypes */
 	FI_DATATYPE_LAST,
+	/*
+	 * enums for 128-bit integer atomics, existing ordering and
+	 * FI_DATATYPE_LAST preserved for compatabilty.
+	 */
+	FI_INT128 = FI_DATATYPE_LAST,
+	FI_UINT128,
+	FI_DATATYPE_LAST_V2,
```

Comment:
Please no LAST_V2... we don't want to duplicate the LAST problem.

Reply:
I know you want this to go away, but how do we handle the sizes of the 2-dimensional arrays in util_atomic.c and ubertest?

Reply:
All users should convert to using the size of an array, like is done in this PR.  We don't want LAST, LAST_v2, LAST_V3, LAST_V4, etc. -- none of which are actually the last value!

We tried being clever by defining LAST.  That made looping easy.  But that loop only works when all the values being looped over handled.  That's why we can't just change the value of LAST - doing so breaks the app if it is re-compiled.  For the same reason, an app can't just loop to LAST_V2.  It has to implement all the values in the range (INT128 and UINT128) before it can.  Once it does that it can iterate over the array, or iterate from INT8 to UINT128.  If we add FLOAT128 or INT256 someday, the app has to change anyway to test it.  Trying to abstract the last value doesn't help.

Reply:
I can certainly remove, but internal to libfabric how do I handle something like this:
extern void (*ofi_atomic_write_handlers[OFI_WRITE_OP_CNT][FI_DATATYPE_LAST])
                        (void *dst, const void *src, size_t cnt);
FI_UINT128 + 1?

Reply:
Either sizeof(ofi_datatype_size_table), FI_UINT128 + 1, or define OFI_DATATYPE_LAST = FI_UINT128 + 1 internally.  Just don't expose it through the stable API.